rdr_service/dao/genomics_dao.py

```diff
@@ -2213,6 +2213,106 @@ def get_report_state_query_config(self):
+        timestamp = pytz.utc.localize(payload_dict.get('date'))
+        appointments = []
+
+        if not payload_dict.get('data'):
+            return {
+                "data": appointments,
+                "timestamp": timestamp
+            }
+
+        for appointment in payload_dict.get('data'):
+            appointments.append({
+                'module': appointment.module_type,
+                'type': 'appointment',
+                'status': appointment.event_type.split('_')[-1],
+                'appointment_id': appointment.appointment_id,
+                'participant_id': f'P{appointment.participant_id}',
+                'note_available': appointment.has_appointment_note
+            })
+
+        return {
+            "data": appointments,
+            "timestamp": timestamp
+        }
+
+    def get_latest_scheduling_data(self, participant_id=None, start_date=None, end_date=None, module=None):
+        max_appointment_id_subquery = sqlalchemy.orm.Query(
+            functions.max(GenomicAppointmentEvent.appointment_id).label(
+                'max_appointment_id'
+            )
+        ).filter(
+            GenomicAppointmentEvent.event_type.notlike('%note_available')
+        ).group_by(
+            GenomicAppointmentEvent.participant_id,
+            GenomicAppointmentEvent.module_type
+        ).subquery()
+        max_event_authored_time_subquery = sqlalchemy.orm.Query(
+            functions.max(GenomicAppointmentEvent.event_authored_time).label(
+                'max_event_authored_time'
+            )
+        ).filter(
+            GenomicAppointmentEvent.event_type.notlike('%note_available')
+        ).group_by(
+            GenomicAppointmentEvent.participant_id,
+            GenomicAppointmentEvent.module_type
+        ).subquery()
+        note_alias = aliased(GenomicAppointmentEvent)
+        with self.session() as session:
+            records = session.query(
+                GenomicAppointmentEvent.participant_id,
+                GenomicAppointmentEvent.module_type,
+                GenomicAppointmentEvent.event_type,
+                GenomicAppointmentEvent.appointment_id,
+                sqlalchemy.case(
+                    [
+                        (note_alias.id.isnot(None), True)
+                    ],
+                    else_=False
+                ).label('has_appointment_note'),
+            ).outerjoin(
+                note_alias,
+                and_(
+                    note_alias.appointment_id == GenomicAppointmentEvent.appointment_id,
+                    note_alias.event_type.like('%note_available')
+                )
+            ).filter(
+                and_(
+                    GenomicAppointmentEvent.appointment_id == max_appointment_id_subquery.c.max_appointment_id,
+                    GenomicAppointmentEvent.event_authored_time ==
+                    max_event_authored_time_subquery.c.max_event_authored_time
+                )
```

Comment:
Do these conditions work if more than one row is returned in the sub-query?  I'm more familiar with using raw SQL and doing an inner join to the sub-query, but it may be handled differently via the ORM.

Reply:
Good question, yes they do I added unit tests to encompass all the usual edge cases to test the queries correctly. Also for each query change, I extracted and ran the corresponding RAW SQL output to make sure the query was being translated correctly from the ORM.

Reply:
I see you're testing multiple participants in `test_pass_start_date_params`.